### PR TITLE
Removes "strandling" status effect from a bunch of random places / refactors it

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Refactors `/datum/status_effect/strandling` to utilize signals so that it's not present in a bunch of random places via hard checks and traits. 

## Why It's Good For The Game

Cleaner code, encapsulation

## Changelog

:cl: Melbert
refactor: Refactored the Durathread golem's strangle status effect
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
most informative changelog

**Comment by MrMelbert** [comment]
is this because i didn't use punctuation

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -402,7 +402,11 @@
 		to_chat(M, span_warning("You can't put [p_them()] out with just your bare hands!"))
 		return
 
-	if(M == src && check_self_for_injuries())
+	if(SEND_SIGNAL(src, COMSIG_CARBON_PRE_HELP_ACT, M) & COMPONENT_BLOCK_HELP_ACT)
+		return
+
+	if(M == src)
```
Comment (tralezab) [inline code/modules/mob/living/carbon/carbon_defense.dm:408]: if you use the var, could you improve it while you're here?

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -693,28 +693,16 @@
...
 		return
 
-	if(src == M)
-		if(has_status_effect(/datum/status_effect/strandling))
-			to_chat(src, span_notice("You attempt to remove the durathread strand from around your neck."))
-			if(do_after(src, 3.5 SECONDS, src))
-				to_chat(src, span_notice("You succesfuly remove the durathread strand."))
-				remove_status_effect(/datum/status_effect/strandling)
-			return
-		check_self_for_injuries()
```
Comment (tralezab) [inline code/modules/mob/living/carbon/human/human_defense.dm:707]: what the f- this was in human defense?!

File: code/datums/status_effects/debuffs/strandling.dm [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+	// Play another sound after we're done
+	tool?.play_tool_sound(owner)
+
+	user.visible_message(
+		span_notice("[user] successfully [tool ? "cut":"remove"] the strand from around [owner == user ? "[owner.p_their()]":"[owner]'s"] neck."),
+		span_notice("You successfully [tool ? "cut":"remove"] the strand from around [owner == user ? "your":"[owner]'s"] neck."),
+	)
+	qdel(src)
+	return TRUE
```
Comment (tralezab) [inline code/datums/status_effects/debuffs/strandling.dm:84]: I'm really kinda meh about checking tool's existence 8 times in this proc but honestly i don't have a better solution so this is fine

**Review by tralezab — CHANGES_REQUESTED** [review]
great work, another codebase sin solved

File: code/modules/mob/living/carbon/human/human_defense.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/mob/living/carbon/human/human_defense.dm:707]: golem moment

**Review by tralezab — APPROVED** [review]

**Pull request merged**